Shared/Supporting Files/Extensions/View+Sheet.swift

```diff
@@ -284,46 +284,47 @@ private struct Sheet<Content>: UIViewRepresentable where Content: View {
         
         // Ensures that the device's layout is such that a sheet should be presented and
         // the hosting controller's sheet presentation controller exists.
         guard isSheetLayout,
               let sheet = model.hostingController.sheetPresentationController else {
             // Dismisses the sheet if it is being presented and a popover should
             // be presented instead.
-            if isPresentedControllerHostingType && !model.hostingController.isBeingDismissed {
+            if presentedControllerIsHosting && !hostingControllerIsBeingDismissed {
                 rootViewController.dismiss(animated: false)
             }
             return
         }
         
-        /// A Boolean value indicating whether the sheet was already presenting.
-        let wasPresenting = rootViewController.presentedViewController != nil
-        
-        /// A Boolean value indicating whether the hosting controller is being dismissed.
-        let isHostBeingDismissed = model.hostingController.isBeingDismissed
-        
-        /// A Boolean value indicating whether the presented view controller is an alert controller.
-        let isPresentedControllerAlertType = rootViewController.presentedViewController is UIAlertController
-        
         if isPresented && !wasPresenting {
             // Sets the sheet presentation controller's delegate.
             sheet.delegate = context.coordinator
             // Configures the hosting controller's sheet presentation controller.
             configureSheetPresentationController(sheet)
             // Presents the hosting controller.
-            rootViewController.present(model.hostingController, animated: model.isTransitioningFromPopover ? false : true)
-        } else if !isPresented && wasPresenting && !isHostBeingDismissed && !isPresentedControllerAlertType {
+            rootViewController.present(model.hostingController, animated: !model.isTransitioningFromPopover)
+        } else if !isPresented && wasPresenting && !hostingControllerIsBeingDismissed && !presentedControllerIsAlert {
             // Dismisses the view controller presented by the root view controller
             // if 'isPresented' is false, but was presenting before (popover), is
             // not currently being dismissed, and is not an alert.
-            rootViewController.dismiss(animated: isPresentedControllerHostingType ? true : false)
-            model.isTransitioningFromPopover = !isPresentedControllerHostingType
-        } else if isHostBeingDismissed {
+            rootViewController.dismiss(animated: presentedControllerIsHosting)
+            model.isTransitioningFromPopover = !presentedControllerIsHosting
+        } else if hostingControllerIsBeingDismissed {
             // Sets 'isPresented' to false when the hosting controller is being dismissed.
-            isPresented = false
-            model.isTransitioningFromPopover = !isPresentedControllerHostingType
+            Task {
```

Comment:
I wasn't able to reproduce this on iPhone, only on iPad. Can you verify this is fixed now?

Reply:
I'm still seeing this occasionally (1 in 10) but since it doesn't cause the app to be in an unusable state, I think I'll settle with what we have now.

Reply:
Upping the time on the `Task.sleep` may fix that. I just tried to go with the minimum value where I stopped seeing the issue, but it's so small that there probably a lot of wiggle room before it is visually noticeable.